src/Microsoft.OData.Core/ODataBatchUtils.cs

```diff
@@ -183,11 +183,7 @@ internal static void EnsureArraySize(ref byte[] buffer, int numberOfBytesInBuffe
             ODataInputContext inputContext)
         {
             Debug.Assert(inputContext != null, "inputContext != null");
-            if (!(inputContext is ODataJsonLightInputContext))
-            {
-                return;
-            }
-            else
+            if (inputContext is ODataJsonLightInputContext)

```

Comment:
The main PR has been updated to use different logic... will skip this one since here only contains non-functional changes.